packages/hoppscotch-app/components/http/ImportCurl.vue

```diff
@@ -116,7 +129,64 @@ const handleImport = () => {
       }
     }
 
-    const method = parsedCurl.method.toUpperCase()
+    const method = parsedCurl.method?.toUpperCase() || "GET"
+    const contentType = parsedCurl.contentType
+
+    // TODO: implement all other types of auth
+    // in a separate helper file
+    // >> preference order:
+    //      - Auth headers
+    //      - --user arg
+    //      - Creds provided along with URL
+    let auth = {} as HoppRESTAuth
+    const user: string = parsedCurl.user ?? ""
+
+    if (parsedCurl.auth) {
+      const { type, token } = parsedCurl.auth
+      auth = {
+        authType: type.toLowerCase(),
+        token,
+      } as HoppRESTAuth
+    } else {
+      auth = {
+        authType: username.length > 0 || user.length > 0 ? "basic" : "none",
+        authActive: true,
+        ...(username.length > 0 && {
+          username,
+          password,
+        }),
+        ...(user.length > 0 && {
+          username: user.split(":")[0],
+          password: user.split(":")[1],
+        }),
+      } as HoppRESTAuth
+    }
```

Comment:
Split this into a separate function ?

Keeping common logic isolated allows us to iterate more quicker

Reply:
Do you mind any suggestions on where I should place the function?

Reply:
Well, I mean like, this logic is pretty self contained right, so I was thinking more like, take this entire bit out and make it as a function in the same file itself. So basically it abstracts the whole Auth parsing bits into a single function.